addons/mail_client_extension/controllers/main.py

```diff
@@ -243,9 +273,50 @@ def res_partner_create(self, email, name, company, **kwargs):
             'name': name,
             'email': email,
         }
-        if company > -1:
+        if company and company > -1:
             partner_info['parent_id'] = company
         partner = request.env['res.partner'].create(partner_info)
 
         response = {'id': partner.id}
         return response
+
+    @http.route('/mail_client_extension/log_mail_content', type="json", auth="outlook", cors="*")
+    def log_mail_content(self, model, res_id, message):
+        assert model in self._get_loggable_models()
+        record = request.env[model].browse(res_id)
+        record.message_post(body=message)
+
+    def _get_loggable_models(self):
```

Comment:
The name is a bit confusing.
How about "_mail_content_logging_models_whitelist" ?

It's a bit verbose, maybe Thibault will have a better suggestion later.
Please add a docstring here as well to explain the purpose and why it's used.